pkg/controller/provider/controller.go

```diff
@@ -354,6 +387,22 @@ func (r *Reconciler) getSecret(provider *api.Provider) (*v1.Secret, error) {
 	return secret, nil
 }
 
+func (r *Reconciler) handleServerCreationFailure(provider *api.Provider, errMsg error) {
```

Comment:
```suggestion
func (r *Reconciler) handleServerCreationFailure(provider *api.Provider, err error) {
```
errMsg is a bit confusing here - we get an error, not a message